### ISSUE TITLE
using shaders with optifine will break locking on [BUG]

### ISSUE THREAD

**Issue by derplordofdeath** [Laike-Endaril/Dynamic-Stealth#125]
i have ZERO other mods on other than optifine and what is required to run dynamic stealth (fantasticLib) and turning on shaders breaks it internal as well if you turn off shaders your fine

shaders off
![no shaders](https://user-images.githubusercontent.com/75479104/101240940-63660200-36a7-11eb-8b0b-bc4e12053eec.png)
![no shaders locking on](https://user-images.githubusercontent.com/75479104/101240941-64972f00-36a7-11eb-9fb9-73909174bc13.png)

heres shaders on
![shaders on](https://user-images.githubusercontent.com/75479104/101240943-67921f80-36a7-11eb-8329-9da411667542.png)
![shaders on locking on](https://user-images.githubusercontent.com/75479104/101240953-68c34c80-36a7-11eb-8e27-079ed1394430.png)

**Comment by Laike-Endaril** [comment]
![image](https://user-images.githubusercontent.com/21181839/101258257-31b85f80-36d6-11eb-8d29-186b1beddfe6.png)
![image](https://user-images.githubusercontent.com/21181839/101258263-3846d700-36d6-11eb-8cd2-420331cc34de.png)

I'm not running into this issue so far...
Can I see your DS config (for both server and client, if not in SP mode)?

**Comment by Laike-Endaril** [comment]
Ah right, here are some shots of the forge/OF/shader versions:
![image](https://user-images.githubusercontent.com/21181839/101258315-91166f80-36d6-11eb-992d-1d1f438421d1.png)
![image](https://user-images.githubusercontent.com/21181839/101258320-9673ba00-36d6-11eb-9c28-d965b9c868e4.png)

**Comment by derplordofdeath** [comment]
I only personal use sildurs vibrant shaders I haven't tested other shaders also gimme a minute

**Comment by derplordofdeath** [comment]
[1.12.2.095+.TXT](https://github.com/Laike-Endaril/Dynamic-Stealth/files/5648687/1.12.2.095%2B.TXT)
heres the config but ill tell you what it is definitely optifine i think its not up to date?

**Comment by derplordofdeath** [comment]
hey guess whos back 
![versions](https://user-images.githubusercontent.com/75479104/101279397-dee2b480-3776-11eb-9320-4e058d55c629.png)
here's my versions sorry for spamming comments lol

**Comment by Laike-Endaril** [comment]
The screenshot is cut off on the right side; I can't tell which OF version you have!

I'll try Sildur's, though I'm assuming that's not it since you mentioned internal shaders as well...

**Comment by derplordofdeath** [comment]
here you go 
![versions 2 0](https://user-images.githubusercontent.com/75479104/101284649-01d09100-3796-11eb-9f51-0540759e028d.png)
yea I fucked up my crop my b

**Comment by Laike-Endaril** [comment]
Huh...looks like we have the same OF version, and I just ran a test with your DS config as well and did not see the issue...

Still with BSL though.  I'll try Sildur's just in case.

Does the issue happen all the time, or just sometimes?

Also, send me your OF and shader options when you get a chance (in main modpack folder; "optionsof.txt", "optionsshaders.txt"

**Comment by derplordofdeath** [comment]
Ok these are my exact settings
sildurs shader settings
[Sildurs Vibrant Shaders v1.27 Lite.zip.txt](https://github.com/Laike-Endaril/Dynamic-Stealth/files/5649043/Sildurs.Vibrant.Shaders.v1.27.Lite.zip.txt)
and the others you requested
[optionsof.txt](https://github.com/Laike-Endaril/Dynamic-Stealth/files/5649044/optionsof.txt)
[optionsshaders.txt](https://github.com/Laike-Endaril/Dynamic-Stealth/files/5649045/optionsshaders.txt)

also to answer your question it happens no matter what every time I play

**Comment by Laike-Endaril** [comment]
Alright, good to know.  That means I can load up and let just one entity get targeted to test correctly.

Just tried Sildurs and it straight up crashed though...same version you have.  It seems to be triggering some OF code that references a non-existing field (field_191308_b; checked the MC mappings and said field truly does not exist, at least not in 1.12.2).  Did you run into this and fix it already, or did this not happen for you?

Also, this might sound crazy, but if you can test the exact same setup on another machine...just in case...
By which I mean the minimum setup needed to produce the issue (you said you still got it to happen with just DS and OF)

**Comment by derplordofdeath** [comment]
I'm not quite sure what issue your talking about never messed around with that stuff also I cant try it on another pc I only have one might be able to get a friend to help test also I'm using "Sildurs Vibrant Shaders v1.27 Lite" I'm not sure if I can copy paste the shader or if that's a TOS violation so your on your own with finding it

**Comment by derplordofdeath** [comment]
just realized it was the file name lol

**Comment by Laike-Endaril** [comment]
Yeah, I made sure I was using the same version (1.27 Lite)...not sure why it's crashing for me and not for you.

Also not sure what you mean by "it was the file name"?

**Comment by Laike-Endaril** [comment]
AHA, got Sildur's to run...had to go into it and delete the "entity.properties" file.

Also reproduced the issue finally.  Looking into it more now.

**Comment by Laike-Endaril** [comment]
Go to your shader options and turn "render quality" to 1x.  It will fix the issue.

I'll see if I can work around OF's transforms on my end.

**Comment by derplordofdeath** [comment]
AH sorry i didnt get back to you i kinda forgot to check in my fault. a quick update, setting it to 1 fixed the issue but it would be awesome if you could fix it asap because personally my pc is... um... dog shit? so it lags a bit but none the less. thank you for the help! 

P.S i have no idea how to handle this github issue stuff so ill leave it to you.

**Comment by Laike-Endaril** [comment]
Np, I plan to leave this open until I at least decide whether or not I can work around this on my end...I think it should be doable, at least.  Just not straightforward.

**Comment by Laike-Endaril** [comment]
This will be fixed in the next version when released (114)
